core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -423,21 +423,23 @@ public void commit() {
     try {
       LOG.info("Committed snapshot {} ({})", newSnapshotId.get(), getClass().getSimpleName());
 
-      // at this point, the commit must have succeeded. after a refresh, the snapshot is loaded by
-      // id in case another commit was added between this commit and the refresh.
-      Snapshot saved = ops.refresh().snapshot(newSnapshotId.get());
-      if (saved != null) {
-        cleanUncommitted(Sets.newHashSet(saved.allManifests(ops.io())));
-        // also clean up unused manifest lists created by multiple attempts
-        for (String manifestList : manifestLists) {
-          if (!saved.manifestListLocation().equals(manifestList)) {
-            deleteFile(manifestList);
+      if (!canSkipCleanupAfterCommitSuccess()) {
```

Comment:
See above, I think this becomes more readable if you invert it, you can just do `if (cleanupAfterCommit())`